contract_sale_generation/models/contract_line.py

```diff
@@ -11,7 +11,7 @@ def _prepare_sale_line_vals(self, dates, order_id=False):
         sale_line_vals = {
             "product_id": self.product_id.id,
             "product_uom_qty": self._get_quantity_to_invoice(*dates),
-            "product_uom": self.uom_id.id,
+            "product_uom": self.product_id.uom_id.id,
```

Comment:
Why changing this ? If we want to use a particular uom on contract line, that one should be used.